apps/api/src/app/services/sights/sights.service.ts

```diff
@@ -52,9 +48,16 @@ export class SightsService {
       .collection(this.collectionName)
       .doc(id)
       .delete();
-    if (!snapshot) {
-      return Promise.resolve(false);
-    }
-    return Promise.resolve(true);
+    return Promise.resolve(!!snapshot);
+  }
+
+  public async getSightByCoordinates(coordinate: Coordinates): Promise<Sight> {
```

Comment:
Try to take out the duplicate code in a separate method. I mean getSightByCoordinates and getSightByID